### PR TITLE
ENH: Update timestamp parsing logic

### PR DESCRIPTION
**Pull request by NPann**
* Remove undefined tag (StudyDateTime)
* Add parsing SeriesTime and SerieDate 
* Update logic order
* Add default time to 00:00 when time is not defined
* Extend test coverage

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -105,44 +105,66 @@ def get_timestamp(dcm, timezone):
     """
     Parse Study Date and Time, return acquisition and session timestamps
     """
+    # Study Date and Time, with precedence as below
     if hasattr(dcm, 'StudyDate') and hasattr(dcm, 'StudyTime') and \
        getattr(dcm, 'StudyDate') and getattr(dcm, 'StudyTime'):
         study_date = dcm.StudyDate
         study_time = dcm.StudyTime
-    elif hasattr(dcm, 'StudyDateTime') and \
```
Comment (kalebdfischer) [inline run.py:112]: Guessing that 'StudyDateTime' is not a real tag.

File: run.py [hunk 2]
```diff
@@ -105,44 +105,66 @@ def get_timestamp(dcm, timezone):
     """
     Parse Study Date and Time, return acquisition and session timestamps
     """
+    # Study Date and Time, with precedence as below
     if hasattr(dcm, 'StudyDate') and hasattr(dcm, 'StudyTime') and \
        getattr(dcm, 'StudyDate') and getattr(dcm, 'StudyTime'):
```
Comment (kalebdfischer) [inline run.py:110]: the get/hasattr can be simplified by providing None as a third parameter to getattr

File: run.py [hunk 3]
```diff
@@ -105,44 +105,66 @@ def get_timestamp(dcm, timezone):
     """
     Parse Study Date and Time, return acquisition and session timestamps
```
Comment (kalebdfischer) [inline run.py:106]: Would be good to capture/summarize the order/priority of Dates/Times used.

File: run.py [hunk 4]
```diff
@@ -105,44 +105,66 @@ def get_timestamp(dcm, timezone):
...
+    elif hasattr(dcm, 'AcquisitionDate') and getattr(dcm, 'AcquisitionDate'):
+        acquisition_date = dcm.AcquisitionDate
+        acquisition_time = '000000.00'
+    elif hasattr(dcm, 'SeriesDate') and getattr(dcm, 'SeriesDate'):
+        acquisition_date = dcm.SeriesDate
+        acquisition_time = '000000.00'
+    elif hasattr(dcm, 'StudyDate') and getattr(dcm, 'StudyDate'):
+        acquisition_date = dcm.StudyDate
+        acquisition_time = '000000.00'
+
```
Comment (kalebdfischer) [inline run.py:167]: Should there be logic to use acquisition_date/time if they are defined and study_date/time are not?

File: tests/unit_tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import pytest
+from pathlib import Path
+
+import pydicom
+from pydicom.data import get_testdata_files
+
+
+@pytest.fixture(scope='function')
+def timestamp_clean_dcm():
```
Comment (kalebdfischer) [inline tests/unit_tests/conftest.py:9]: nice fixture

File: tests/unit_tests/test_get_timestamps.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+import tempfile
+import pytz
+import tzlocal
+from pathlib import Path
+from pydicom.data import get_testdata_files
+
+from run import get_timestamp, validate_timezone
+
+
+def test_get_timestamp_logic(timestamp_clean_dcm):
```
Comment (kalebdfischer) [inline tests/unit_tests/test_get_timestamps.py:11]: that's the stuff

File: run.py (hunk 1 above)
Comment (NPann) [reply run.py:112]: Correct

File: run.py (hunk 2 above)
Comment (NPann) [reply run.py]: b4f17a7

File: run.py (hunk 4 above)
Comment (NPann) [reply run.py:166]: 444865c

File: run.py (hunk 3 above)
Comment (NPann) [reply run.py]: 9d6b15a

**Pull request merged**